frontend/pages/admin/TeamManagementPage/TeamDetailsWrapper/TeamSettings/TeamSettings.tsx

```diff
@@ -0,0 +1,197 @@
+import React, { useCallback, useContext, useEffect, useState } from "react";
+import { useQuery } from "react-query";
+import teamsAPI, { ILoadTeamResponse } from "services/entities/teams";
+import configAPI from "services/entities/config";
+import { IConfig } from "interfaces/config";
+import { NotificationContext } from "context/notification";
+import { IApiError } from "interfaces/errors";
+import Spinner from "components/Spinner";
+import DataError from "components/DataError";
+import TeamHostExpiryToggle from "./components/TeamHostExpiryToggle";
+
+const baseClass = "team-settings";
+
+const HOST_EXPIRY_ERROR_TEXT = "Host expiry window must be a positive number.";
+
+const TeamSettings = ({ location, router }: ITeamSubnavProps) => {
+  const [UITeamHostExpiryEnabled, setUITeamHostExpiryEnabled] = useState(false); // default false until API response
+  const [UITeamHostExpiryWindow, setUITeamHostExpiryWindow] = useState<
+    number | string
+  >("");
+  const [updatingTeamSettings, setUpdatingTeamSettings] = useState(false);
+  const [formErrors, setFormErrors] = useState<Record<string, string | null>>(
+    {}
+  );
+  const [addingCustomWindow, setAddingCustomWindow] = useState(false);
+
+  const { renderFlash } = useContext(NotificationContext);
+
+  const { isRouteOk, teamIdForApi } = useTeamIdParam({
+    location,
+    router,
+    includeAllTeams: false,
+    includeNoTeam: false,
+    permittedAccessByTeamRole: {
+      admin: true,
+      maintainer: false,
+      observer: false,
+      observer_plus: false,
+    },
+  });
+
+  const {
+    data: appConfig,
+    isLoading: isLoadingAppConfig,
+    error: errorLoadGlobalConfig,
+  } = useQuery<IConfig, Error, IConfig>(
+    ["globalConfig"],
+    () => configAPI.loadAll(),
+    { refetchOnWindowFocus: false }
+  );
+  const {
+    host_expiry_settings: {
+      host_expiry_enabled: globalHostExpiryEnabled,
+      host_expiry_window: globalHostExpiryWindow,
+    },
+  } = appConfig ?? { host_expiry_settings: {} };
+
+  const {
+    isLoading: isLoadingTeamConfig,
+    refetch: refetchTeamConfig,
+    error: errorLoadTeamConfig,
+  } = useQuery<ILoadTeamResponse, Error, ITeamConfig>(
+    ["teamConfig", teamIdForApi],
+    () => teamsAPI.load(teamIdForApi),
+    {
+      enabled: isRouteOk && !!teamIdForApi,
+      select: (data) => data.team,
+      onSuccess: (teamConfig) => {
+        // default this setting to current team setting
+        // can be updated by user actions
+        setUITeamHostExpiryEnabled(
+          teamConfig?.host_expiry_settings?.host_expiry_enabled ?? false
+        );
+        setUITeamHostExpiryWindow(
+          teamConfig.host_expiry_settings?.host_expiry_window ?? ""
+        );
+      },
+      refetchOnWindowFocus: false,
+    }
+  );
+
+  const validate = useCallback(() => {
+    const errors: Record<string, string> = {};
+    if (
+      (!globalHostExpiryEnabled &&
+        UITeamHostExpiryEnabled &&
+        !UITeamHostExpiryWindow) ||
```

Comment:
So this field can be empty if the expiry is not enabled, correct?

Reply:
It can be empty if there is a _global_ expiry set, in which case it should disable the local setting. Otherwise it can't be empty – [devnote here](https://www.figma.com/file/V0gEWDC3j50ZAadgEtyp4A/%2315609-Configure-a-different-host-expiry-per-team?type=design&node-id=542%3A1687&mode=design&t=EvxYslPlNCfZZGuX-1).

Reply:
It's not working atm, pending coordination with @getvictor about the expected zero value in that case

Reply:
Currently sending `0` when the field is empty, which the API is not respecting. May change to a different zero value pending that discussion.

Reply:
There are only 2 states that team setting supports:
- `false` picks up the global setting
- `true` picks up the local window